blog/models.py

```diff
@@ -81,6 +82,11 @@ def image(self):
         if len(self.images):
             return self.images[0]
 
+    def delete(self):
```

Comment:
Because you are overriding the `delete` method that is native, you're not sure what it arguments are passing. You should therefore relay them:
```def delete(self, *args, **kwargs)```

You're looping into a recursion when calling `delete()` in the `delete()` method.
When overriding a method you should anyways call the parent method that does a bunch of things:

```
super(Article, self).delete(*args, **kwargs)
```